ddtrace/contrib/kafka/patch.py

```diff
@@ -229,13 +191,6 @@ def traced_commit(func, instance, args, kwargs):
     if not pin or not pin.enabled():
         return func(*args, **kwargs)
 
-    if config._data_streams_enabled:
-        message = get_argument_value(args, kwargs, 0, "message")
-        offsets = kwargs.get("offsets", [])
-        if message is not None:
-            offsets = [TopicPartition(message.topic(), message.partition(), offset=message.offset())]
-        for offset in offsets:
-            pin.tracer.data_streams_processor.track_kafka_commit(
-                instance._group_id, offset.topic, offset.partition, offset.offset or -1, time.time()
-            )
+    core.dispatch("kafka.commit.start", [instance, args, kwargs])
```

Comment:
are we sure the consumer of this event will be executed before the function is called?

Reply:
The calls to core and events are (and will be for the foreseeable future) synchronous.  APM, DSM, and ASM all have the same requirements to be able to modify parameters and take action prior to execution, so its a necessary behavior